.github/workflows/tests.yml

```diff
@@ -406,21 +370,92 @@ jobs:
           ls -lhR benchmarks
           bash ci/scripts/combine-dbs.sh
 
-      - name: Upload benchmark db
-        env:
-          AWS_ACCESS_KEY_ID: ${{ secrets.RUNTIME_CI_BOT_AWS_ACCESS_KEY_ID }}
-          AWS_SECRET_ACCESS_KEY: ${{ secrets.RUNTIME_CI_BOT_AWS_SECRET_ACCESS_KEY }}
-          AWS_DEFAULT_REGION: us-east-2  # this is needed for boto for some reason
-          DB_NAME: benchmark.db
-        run: |
-          aws s3 cp $DB_NAME s3://coiled-runtime-ci/benchmarks/
+      # - name: Upload benchmark db
+      #   env:
+      #     AWS_ACCESS_KEY_ID: ${{ secrets.RUNTIME_CI_BOT_AWS_ACCESS_KEY_ID }}
+      #     AWS_SECRET_ACCESS_KEY: ${{ secrets.RUNTIME_CI_BOT_AWS_SECRET_ACCESS_KEY }}
+      #     AWS_DEFAULT_REGION: us-east-2  # this is needed for boto for some reason
+      #     DB_NAME: benchmark.db
+      #   run: |
+      #     aws s3 cp $DB_NAME s3://coiled-runtime-ci/benchmarks/
 
       - name: Upload benchmark results as artifact
         uses: actions/upload-artifact@v3
         with:
           name: benchmark.db
           path: benchmark.db
 
+  regressions:
+    needs: process-results
+    # Always check for regressions, as this can be skipped even if an indirect dependency fails (like a test run)
+    if: always() && github.ref == 'refs/heads/main' && github.repository == 'coiled/coiled-runtime'
+    name: Detect regressions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: benchmark.db
+
+      - name: Set up environment
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          use-mamba: true
+          python-version: "3.9"
+          environment-file: ci/environment-dashboard.yml
+
+      - name: Run detect regressions
+        run: |
+          python detect_regressions.py
+      
+      - name: Create regressions summary
+        if: always() && github.ref == 'refs/heads/main' && github.repository == 'coiled/coiled-runtime'
```

Comment:
I don't think we need the `ref` or `repository` check here

Reply:
do we want to run this on PRs too? I thought since we are only updating the database on ref = main, we should keep this. Or will this be cover since the check it's also at the step level.

Reply:
Isn't this job gated by the same check above? I just mean that it's redundant.

That being said, I actually think that running this on PRs would be good practice. We mostly just don't want to upload the new db on PRs